### PR TITLE
Build with vulkan backend source files and add corlor format mapping from gbm to vulkan

### PR DESCRIPTION
**Pull request by gaochun**
Keep the session at https://github.com/intel/IA-Hardware-Composer/pull/108, @kalyankondapally Can you see the latest update in the PR?

### CODE REVIEW

File: os/android/platformdefines.h [hunk 1]
```diff
@@ -32,6 +32,10 @@
 
 #include <ui/GraphicBuffer.h>
 
+#ifdef USE_VK
+#include <gbm.h>
```
Comment (kalyankondapally) [inline os/android/platformdefines.h]: Hmm why do I Still see it here ?

Comment (kalyankondapally) [inline os/android/platformdefines.h]: Ha I see u remove in the next patch, but can you already remove it here ?

**Comment by gaochun** [comment]
@kalyankondapally Removed the gbm header from first patch, please review again, thanks.

File: Android.mk [hunk 2]
```diff
@@ -74,6 +74,7 @@ LOCAL_SRC_FILES := \
 	wsi/drm/drmplane.cpp \
 	wsi/drm/drmdisplaymanager.cpp \
```
Comment (kalyankondapally) [inline Android.mk]: typo in commit message "color"

File: Android.mk [hunk 3]
```diff
@@ -116,7 +116,26 @@ ifeq ($(strip $(BOARD_DISABLE_NATIVE_COLOR_MODES)),true)
...
+ifeq ($(strip $(BOARD_USES_VULKAN)),true)
+LOCAL_SHARED_LIBRARIES += \
+	libvulkan
+
+LOCAL_CPPFLAGS += \
+	-DUSE_VK \
+	-DDISABLE_EXPLICIT_SYNC
+
+LOCAL_C_INCLUDES += \
+	$(INTEL_MINIGBM) \
```
Comment (kalyankondapally) [inline Android.mk]: do we need this ?

Comment (gaochun) [inline Android.mk]: Don't need anymore, thanks.

**Comment by gaochun** [comment]
Removed the minigbm include from Android.mk

File: os/android/platformdefines.h (hunk 1 above)
Comment (kalyankondapally) [inline os/android/platformdefines.h]: this came back :)

**Comment by kalyankondapally** [comment]
merged thanks

**Pull request closed without merging**